restli-tools/src/main/resources/apiVmTemplates/action.vm

```diff
@@ -13,7 +13,55 @@ Copyright (c) 2021 LinkedIn Corp.
    See the License for the specific language governing permissions and
    limitations under the License.
 *#
+#if($is_interface)
+  #foreach($method in $spec.actions)
+    #set($actionParamClassName = "${util.nameCapsCase($method.name)}ActionParameters")
+    #set($actionOptionalParamClassName = "${util.nameCapsCase($method.name)}ActionOptionalParameters")
+    #doc($method.schema.doc)
+
+    #if(${method.hasRequiredParams()}) ## action with no required Params will only have one API
+    @SuppressWarnings("unchecked")
+    public CompletionStage<${method.valueClassDisplayName}> ${util.nameCamelCase(${method.name})}(
+      #if(${method.isEntityAction()})
+      $spec.keyClassName $spec.idName,
+      #end
+        #set($paramLen = ${method.getRequiredParameters().size()})
+        #if($paramLen > 0)
+          #set($lastParam = ${method.getRequiredParameters().get($paramLen - 1)})
+          #if($paramLen > 1)
+            #set($endIdx= ${paramLen} - 2)
+            #foreach($idx in [0..$endIdx])
+              #set($param = ${method.getRequiredParameters().get($idx)})
+            $param.paramClassDisplayName $param.paramName,
```

Comment:
use $forEach.hasNext to simplify this :)

Reply:
sure. Already did